File: app/src/main/java/org/ygba/youthgobudget/dialogs/DistrictPickerActivity.java
package org.ygba.youthgobudget.dialogs;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import org.ygba.youthgobudget.R;
import org.ygba.youthgobudget.YGBARepository;
import org.ygba.youthgobudget.data.YGBDatabase;
import org.ygba.youthgobudget.data.helpers.district.District;

import java.util.List;
import java.util.concurrent.ExecutionException;

public class DistrictPickerActivity extends Activity implements DistrictPickerAdapter.OnDistrictClickListener {
    public static final String DISTRICT_NAME = "org.ygba.youthgobudget.dialogs.DistrictPickerActivity.DISTRICT_NAME";
    public static final String DISTRICT_ID = "org.ygba.youthgobudget.dialogs.DistrictPickerActivity.DISTRICT_ID";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_district_picker);
        RecyclerView recyclerView = findViewById(R.id.district_recycler_view);
        DistrictPickerAdapter districtAdapter = new DistrictPickerAdapter(this, getDistricts());
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        recyclerView.setAdapter(districtAdapter);
    }

    private List<District> getDistricts() {
        try {
            return YGBARepository.getInstance(YGBDatabase.getInstance(this)).getDistrictList();
        } catch (ExecutionException | InterruptedException e) {
            e.printStackTrace();
        }
        return null;
    }

    @Override
    public void onDistrictClick(District district) {
        Intent intent = new Intent();
        intent.putExtra(DISTRICT_NAME, district.getName());
        intent.putExtra(DISTRICT_ID, district.getId());
        setResult(RESULT_OK, intent);
        finish();
    }
}

File: app/src/main/java/org/ygba/youthgobudget/dialogs/DistrictPickerActivity.kt
package org.ygba.youthgobudget.dialogs

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.RecyclerView
import org.ygba.youthgobudget.R
import org.ygba.youthgobudget.data.helpers.district.District

class DistrictPickerActivity : AppCompatActivity() {
    private lateinit var districtPickerViewModel: DistrictPickerViewModel;

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_district_picker)

        districtPickerViewModel = ViewModelProviders.of(this).get(DistrictPickerViewModel::class.java)

        val districtRecyclerView: RecyclerView = findViewById(R.id.district_recycler_view);
        val districtAdapter: DistrictAdapter = DistrictAdapter(this);

    }

    class DistrictAdapter(val context: Context): RecyclerView.Adapter<DistrictAdapter.DistrictViewHolder>() {
        lateinit var districts: List<District>;

        class DistrictViewHolder(itemView: View): RecyclerView.ViewHolder(itemView) {
            val textView: TextView = itemView.findViewById(R.id.district_name);
        }

        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DistrictViewHolder {
            val view = LayoutInflater.from(context).inflate(R.layout.item_district, parent, false);
            return DistrictViewHolder(view)
        }

        override fun getItemCount(): Int {
            return districts.size
        }

        override fun onBindViewHolder(holder: DistrictViewHolder, position: Int) {
            holder.textView.text = districts[position].name
        }

        fun loadDistricts(districtList: List<District>) {
            districts = districtList;
            notifyDataSetChanged()
        }
    }
}